scipy/stats/_multivariate.py

```diff
@@ -3521,3 +3522,353 @@ def rvs(self, eigs, random_state=None, tol=1e-13, diag_tol=1e-7):
+        Log of the probability density function.
+    ``rvs(loc=0, variance_scale=1, shape=1, scale=1, size=1, random_state=None)``
+        Draw random samples from normal and inverse gamma distribution.
+    ``mean(loc=0, variance_scale=1, shape=1, scale=1)``
+        Mean of the random variates.
+    ``mode(loc=0, variance_scale=1, shape=1, scale=1)``
+        Mode of the random variates.
+
+    Parameters
+    ----------
+    x : array
+            One-dimensional array.
+    sig2 : array
+            One-dimensional array.
+    %(_nig_doc_default_callparams)s
+    %(_doc_random_state)s
+
+    Alternatively, the object may be called (as a function) to fix the
+    loc, variance_scale, shape, scale parameters, returning a "frozen"
+    normal inverse gamma random variable:
+
+    rv = normal_invgamma(loc=0, variance_scale=1, shape=1, scale=1)
+        - Frozen object with the same methods but holding the given
+          loc, variance_scale, shape, scale.
+
+    Notes
+    -----
+    %(_nig_doc_callparams_note)s
+
+    The probability density function for `normal_invgamma` is
+
+    .. math::
+
+        f(x,\sigma^2) = \frac {1} {\sigma\sqrt{2\pi\nu} } \, \frac{\beta^\alpha}
+                        {\Gamma(\alpha)} \, \left( \frac{1}{\sigma^2} \right)^
+                        {\alpha + 1}\exp \left( \frac { -\beta}{\sigma^2} -
+                        \frac{(x - \mu)^2} {2\sigma^2\nu}  \right),
+
+    where :math:`\mu` is the loc, :math:`\nu` the variance scale,
+    :math:`\alpha` the shape, :math:`\beta` the scale and :math:`x` and
+    `\sigma^2` takes values.
+
+    .. versionadded:: 0.19.0
+
+    Examples
+    --------
+    >>> import matplotlib.pyplot as plt
+    >>> from mpl_toolkits.mplot3d import axes3d
+    >>> from scipy.stats import normal_invgamma
+    >>> x = np.linspace(0, 5, 10, endpoint=False)
+    >>> sig2 = np.linspace(5, 10, 10, endpoint=False)
+    >>> z = normal_invgamma.pdf(x, sig2, loc=2,
+                                    variance_scale=3, shape=5, scale=2); z
+    array([  5.15655226e-06,   3.07181064e-06,   1.87273568e-06,
+         1.16664399e-06,   7.41424234e-07,   4.79916138e-07,
+         3.15923431e-07,   2.11211796e-07,   1.43229349e-07,
+         9.84088858e-08])
+    >>> fig = plt.figure()
+    >>> ax = fig.add_subplot(111, projection='3d')
+    >>> ax.plot_surface(x, sig2, z)
+
+    """   
+    def __init__(self, seed=None):
+        super(normal_invgamma_gen, self).__init__(seed)
+        self.__doc__ = doccer.docformat(self.__doc__, nig_docdict_params)
+
+    def __call__(self, loc=0, variance_scale=1, shape=1, scale=1, seed=None):
+        """
+        Create a frozen normal inverse gamma distribution.
+
+        See `normal_invgamma_frozen` for more information.
+
+        """
+        return normal_invgamma_frozen(loc, variance_scale, shape, scale,
+                                    seed=seed)
+
+    def _check_parameters(self, loc, variance_scale, shape, scale):
```

Comment:
Why should all parameters be scalars? Other distributions allow for array-like parameters, and do broadcasting if that makes sense.